src/adr/0010-enable-elasticsearch-routing.md

```diff
@@ -0,0 +1,41 @@
+# 10 Enable ElasticSearch routing
+
+Date: 2019-10-17
+
+## Status
+
+Proposed
+
+Additional performance testing is required for adoption.
+
+## Context
+
+Our queries are mostly bounded to a mailbox or a user. We can easily
+limit the number of ElasticSearch nodes involved in a given query by
+grouping the underlying documents on the same node using a routing key.
+
+Without routing key, each shard needs to execute the query. The coordinator
+needs also to be waiting for the slowest shard.
+
+Using the routing key unlocks significant throughput enhancement (proportional
+to the number of shard) and also a possible high percentile latencies enhancement.
+This allows to be more linearly scalable.
+
+## Decision
+
+Enable ElasticSearch routing.
+
+Messages should be indexed by mailbox.
```

Comment:
that means a Message that is present in several mailboxes is indexed several time?

Reply:
Yes. And it is already the case.